File: find_closest_value_in_bst/find_closest_value_in_bst.py
def find_closest_value_in_bst(tree, target):
  current_node = tree
  closest_value = current_node.value

  while current_node is not None:
    if target > current_node.value:
      if abs(current_node.value - target) < abs(closest_value - target):
        closest_value = current_node.value
      current_node = current_node.right
    else:
      if abs(current_node.value - target) < abs(closest_value - target):
        closest_value = current_node.value
      current_node = current_node.left
  return closest_value

class BST:
  def __init__(self, value):
      self.value = value
      self.left = None
      self.right = None


bst = BST(10)
bst.left = BST(8)
bst.right = BST(12)

print(find_closest_value_in_bst(bst, 20))

File: find_closest_value_in_bst/find_closest_value_in_bst.go
package main

import (
    "fmt"
    "math"
)

type BST struct {
    Value int
    Left  *BST
    Right *BST
}

func findClosestValueInBST(tree *BST, target int) int {
    currentNode := tree
    closestValue := currentNode.Value

    for currentNode != nil {
        if abs(currentNode.Value-target) < abs(closestValue-target) {
            closestValue = currentNode.Value
        }
        if target > currentNode.Value {
            currentNode = currentNode.Right
        } else {
            currentNode = currentNode.Left
        }
    }
    return closestValue
}

func abs(x int) int {
    return int(math.Abs(float64(x)))
}

func main() {
    bst := &BST{Value: 10}
    bst.Left = &BST{Value: 8}
    bst.Right = &BST{Value: 12}

    fmt.Println(findClosestValueInBST(bst, 20))
}
